src/core/ext/transport/chttp2/transport/parsing.cc

```diff
@@ -409,67 +409,75 @@ static void on_initial_header(void* tp, grpc_mdelem md) {
     gpr_free(value);
   }
 
-  if (grpc_slice_eq(GRPC_MDKEY(md), GRPC_MDSTR_GRPC_STATUS) &&
-      !grpc_mdelem_eq(md, GRPC_MDELEM_GRPC_STATUS_0)) {
-    /* TODO(ctiller): check for a status like " 0" */
-    s->seen_error = true;
-  }
+  if (GRPC_MDELEM_STORAGE(md) == GRPC_MDELEM_STORAGE_STATIC) {
+    if (md.payload == GRPC_MDELEM_GRPC_STATUS_1.payload ||
+        md.payload == GRPC_MDELEM_GRPC_STATUS_2.payload) {
+      s->seen_error = true;
+    }
+  } else {
+    if (grpc_slice_eq(GRPC_MDKEY(md), GRPC_MDSTR_GRPC_STATUS) &&
+        !grpc_mdelem_eq(md, GRPC_MDELEM_GRPC_STATUS_0)) {
+      /* TODO(ctiller): check for a status like " 0" */
+      s->seen_error = true;
+    }
 
-  if (grpc_slice_eq(GRPC_MDKEY(md), GRPC_MDSTR_GRPC_TIMEOUT)) {
-    grpc_millis* cached_timeout =
-        static_cast<grpc_millis*>(grpc_mdelem_get_user_data(md, free_timeout));
-    grpc_millis timeout;
-    if (cached_timeout != nullptr) {
-      timeout = *cached_timeout;
-    } else {
-      if (GPR_UNLIKELY(
-              !grpc_http2_decode_timeout(GRPC_MDVALUE(md), &timeout))) {
-        char* val = grpc_slice_to_c_string(GRPC_MDVALUE(md));
-        gpr_log(GPR_ERROR, "Ignoring bad timeout value '%s'", val);
-        gpr_free(val);
-        timeout = GRPC_MILLIS_INF_FUTURE;
+    if (grpc_slice_eq(GRPC_MDKEY(md), GRPC_MDSTR_GRPC_TIMEOUT)) {
```

Comment:
Do you mean this to be inside of the else block above? Isn't that changing the behavior? Before this would be executed un conditionally, now it is skipped if `GRPC_MDELEM_STORAGE(md) == GRPC_MDELEM_STORAGE_STATIC` is true

Reply:
I intend it to be skipped because I noticed that none of the grpc static elements have a key of TIMEOUT. So if `GRPC_MDELEM_STORAGE(md) == GRPC_MDELEM_STORAGE_STATIC` is true, we no longer need to go down this code path.

Reply:
Isn't that just a static string? There are no metadata elements with a key of timeout here: https://github.com/grpc/grpc/blob/master/src/core/lib/transport/static_metadata.h#L263

Could a mdelem have `GRPC_MDELEM_STORAGE(md) == GRPC_MDELEM_STORAGE_STATIC` and not be present in the static metadata table?

Reply:
Ah I see what you are saying, makes sense